providers/rxe/rxe.c

```diff
@@ -162,20 +162,144 @@ static int rxe_dereg_mr(struct verbs_mr *vmr)
 	return 0;
 }
 
+static int cq_start_poll(struct ibv_cq_ex *current,
+			 struct ibv_poll_cq_attr *attr)
+{
+	struct rxe_cq *cq = container_of(current, struct rxe_cq, vcq.cq_ex);
+
+	pthread_spin_lock(&cq->lock);
+
+	atomic_thread_fence(memory_order_acquire);
```

Comment:
atomic_thread_fence/atomic_XX pairs should always be merged into an atomic_XX_explicit operation for performance